<a href="https://colab.research.google.com/github/the1975/Data-Processing/blob/main/Learning_Data_Processing/Pandas_Numpy_Task_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Task 4 Data Processing Explanatory Data Analysis dengan Numpy dan Pandas

### Nama : Leonardo yanuarianto
### NIM : 2513020074



lakukan EDA dengan menggunakan dataset dari kaggle dibawah ini

1. **Titanic Dataset** - https://www.kaggle.com/datasets/yasserh/titanic-dataset
2. **Students Performance Datase**t - https://www.kaggle.com/datasets/spscientist/students-performance-in-exams
3. **Superstore Sales Dataset** - https://www.kaggle.com/datasets/vivek468/superstore-dataset-final

**catatan**
-  Setiap mahasiswa WAJIB mengerjakan **minimal 2 dataset**
- **Nilai maksimal** jika mengerjakan 3 dataset lengkap

## Import Library Pandas dan Numpy

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

garis = "=" * 50

Mounted at /content/drive


In [ ]:
import os
print(os.listdir())

import os
print(os.getcwd())

import pandas as pd

import numpy as np

['.config', '.ipynb_checkpoints', 'TitanicDataset.csv', 'SampleSuperstore.csv', 'StudentsPerformance.csv', 'sample_data']
/content


## Load Dataset

load tiap dataset yang digunakan

In [ ]:
# Memuat ketiga dataset.
df_titanic = pd.read_csv('TitanicDataset.csv')
df_students = pd.read_csv('StudentsPerformance.csv')
df_superstore = pd.read_csv('/content/SampleSuperstore.csv', encoding='latin-1')
print(df_superstore.head())
print('berhasil di import')

   Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   Sout

## Data Understanding

lakukan analisis berikut pada tiap dataset
- **Jumlah baris & kolom**
- **Nama kolom**
- **Tipe data**
- **Preview data**

In [ ]:
#Membuat list dataset dan namanya untuk dilooping
datasets = [('Titanic', df_titanic), ('Students Performance', df_students), ('Sample Superstore', df_superstore)]

for name, df in datasets:
    print(f"Data Understanding: {name}\n")
    print(f"Jumlah Baris & Kolom : {df.shape}")
    print(f"Nama Kolom           : {list(df.columns)}")
    print("\nTipe Data:")
    print(df.dtypes)
    print("\nPreview Data (3 baris pertama):")
    display(df.head(3))
    print("-" * 50, "\n")

Data Understanding: Titanic

Jumlah Baris & Kolom : (891, 12)
Nama Kolom           : ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']

Tipe Data:
PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

Preview Data (3 baris pertama):


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S


-------------------------------------------------- 

Data Understanding: Students Performance

Jumlah Baris & Kolom : (1000, 8)
Nama Kolom           : ['gender', 'race/ethnicity', 'parental level of education', 'lunch', 'test preparation course', 'math score', 'reading score', 'writing score']

Tipe Data:
gender                         object
race/ethnicity                 object
parental level of education    object
lunch                          object
test preparation course        object
math score                      int64
reading score                   int64
writing score                   int64
dtype: object

Preview Data (3 baris pertama):


,gender,race/ethnicity,parental level of education,lunch,test preparation course,math score,reading score,writing score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88
2,female,group B,master's degree,standard,none,90,95,93


-------------------------------------------------- 

Data Understanding: Sample Superstore

Jumlah Baris & Kolom : (9994, 21)
Nama Kolom           : ['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode', 'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State', 'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category', 'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit']

Tipe Data:
Row ID             int64
Order ID          object
Order Date        object
Ship Date         object
Ship Mode         object
Customer ID       object
Customer Name     object
Segment           object
Country           object
City              object
State             object
Postal Code        int64
Region            object
Product ID        object
Category          object
Sub-Category      object
Product Name      object
Sales            float64
Quantity           int64
Discount         float64
Profit           float64
dtype: object

Preview Data (3 baris pertama):


,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.96,2,0.0,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.94,3,0.0,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.62,2,0.0,6.8714


-------------------------------------------------- 



**Interpretas tiap dataset (Wajib Diisi)**

1. Dataset berisi tentang apa?
2. Kolom apa saja yang penting?

Jawaban Anda disini

1. **Dataset Titanic**

Data ini berisi tentang data penumpang titanic dan apakah mereka selamat atau tidak.

Kolom penting: Survive, Pclass, Sex, Age, Fare, dan Embarked.

2. **Dataset Student**

Data yang berisi nilai siswa berdasarkan faktor sosial dan pendidikan

Kolom penting: math score, reading score, writing score, race/ethnicity, parental level of education, lunch, test preparation course.

3. **Dataset Superstore**

Data set ini berisi tentang penjualan retail (bisnis)

Kolom penting: Sales, Profit, Discount, Category, Region

## EXPLORATORY DATA ANALYSIS (EDA)

lakukan EDA pada tiap dataset yang digunakan

1. Statistik Deskriptif

In [ ]:
# kode disini
print("Statistik Deskriptif Titanic\n")
display(df_titanic.describe(include='all'))

print("\nStatistik Deskriptif Students Performance\n")
display(df_students.describe(include='all'))

print("\nStatistik Deskriptif Superstore\n")
display(df_superstore.describe(include='all'))

Statistik Deskriptif Titanic



,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
count,891.000000,891.000000,891.000000,891,891,714.000000,891.000000,891.000000,891,891.000000,204,889
unique,NaN,NaN,NaN,891,2,NaN,NaN,NaN,681,NaN,147,3
top,NaN,NaN,NaN,"Dooley, Mr. Patrick",male,NaN,NaN,NaN,347082,NaN,G6,S
freq,NaN,NaN,NaN,1,577,NaN,NaN,NaN,7,NaN,4,644
mean,446.000000,0.383838,2.308642,NaN,NaN,29.699118,0.523008,0.381594,NaN,32.204208,NaN,NaN
std,257.353842,0.486592,0.836071,NaN,NaN,14.526497,1.102743,0.806057,NaN,49.693429,NaN,NaN
min,1.000000,0.000000,1.000000,NaN,NaN,0.420000,0.000000,0.000000,NaN,0.000000,NaN,NaN
25%,223.500000,0.000000,2.000000,NaN,NaN,20.125000,0.000000,0.000000,NaN,7.910400,NaN,NaN
50%,446.000000,0.000000,3.000000,NaN,NaN,28.000000,0.000000,0.000000,NaN,14.454200,NaN,NaN
75%,668.500000,1.000000,3.000000,NaN,NaN,38.000000,1.000000,0.000000,NaN,31.000000,NaN,NaN



Statistik Deskriptif Students Performance



,gender,race/ethnicity,parental level of education,lunch,test preparation course,math score,reading score,writing score
count,1000,1000,1000,1000,1000,1000.00000,1000.000000,1000.000000
unique,2,5,6,2,2,NaN,NaN,NaN
top,female,group C,some college,standard,none,NaN,NaN,NaN
freq,518,319,226,645,642,NaN,NaN,NaN
mean,NaN,NaN,NaN,NaN,NaN,66.08900,69.169000,68.054000
std,NaN,NaN,NaN,NaN,NaN,15.16308,14.600192,15.195657
min,NaN,NaN,NaN,NaN,NaN,0.00000,17.000000,10.000000
25%,NaN,NaN,NaN,NaN,NaN,57.00000,59.000000,57.750000
50%,NaN,NaN,NaN,NaN,NaN,66.00000,70.000000,69.000000
75%,NaN,NaN,NaN,NaN,NaN,77.00000,79.000000,79.000000



Statistik Deskriptif Superstore



,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
count,9994.000000,9994,9994,9994,9994,9994,9994,9994,9994,9994,...,9994.000000,9994,9994,9994,9994,9994,9994.000000,9994.000000,9994.000000,9994.000000
unique,NaN,5009,1237,1334,4,793,793,3,1,531,...,NaN,4,1862,3,17,1850,NaN,NaN,NaN,NaN
top,NaN,CA-2017-100111,9/5/2016,12/16/2015,Standard Class,WB-21850,William Brown,Consumer,United States,New York City,...,NaN,West,OFF-PA-10001970,Office Supplies,Binders,Staple envelope,NaN,NaN,NaN,NaN
freq,NaN,14,38,35,5968,37,37,5191,9994,915,...,NaN,3203,19,6026,1523,48,NaN,NaN,NaN,NaN
mean,4997.500000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,55190.379428,NaN,NaN,NaN,NaN,NaN,229.858001,3.789574,0.156203,28.656896
std,2885.163629,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,32063.693350,NaN,NaN,NaN,NaN,NaN,623.245101,2.225110,0.206452,234.260108
min,1.000000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,1040.000000,NaN,NaN,NaN,NaN,NaN,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,23223.000000,NaN,NaN,NaN,NaN,NaN,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,56430.500000,NaN,NaN,NaN,NaN,NaN,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,90008.000000,NaN,NaN,NaN,NaN,NaN,209.940000,5.000000,0.200000,29.364000


2. Missing Value

In [ ]:
datasets = [('Titanic', df_titanic), ('Students Performance', df_students), ('Superstore', df_superstore)]

for name, df in datasets:
    print(f"Missing Values: {name}\n")
    missing = df.isnull().sum()
    print(missing[missing > 0])
    print(f"Total sel kosong: {df.isnull().sum().sum()}\n")

Missing Values: Titanic

Age         177
Cabin       687
Embarked      2
dtype: int64
Total sel kosong: 866

Missing Values: Students Performance

Series([], dtype: int64)
Total sel kosong: 0

Missing Values: Superstore

Series([], dtype: int64)
Total sel kosong: 0



**Analisis tiap dataset**

1. Apakah ada data kosong?
2. Kolom mana yang bermasalah?

**Dataset Titanic**: Ada data kosong pada kolom Age (177), Cabin (687), dan Embarked (2).

**Dataset Students Performance**: Tidak ada data kosong.

**Dataset Superstore**: Tidak ada data kosong.Jawaban Anda disini

3. Analisis Numerik

- Rerata
- Nilai tertinggi
- nilai terendah

In [ ]:
# Memilih beberapa kolom numerik penting dari tiap dataset untuk dianalisis
print("Analisis Numerik Titanic (Age & Fare)\n")
display(df_titanic[['Age', 'Fare']].agg(['mean', 'max', 'min']))

print("\nAnalisis Numerik Students Performance (Scores)\n")
display(df_students[['math score', 'reading score', 'writing score']].agg(['mean', 'max', 'min']))

print("\nAnalisis Numerik Superstore (Sales & Profit)\n")
display(df_superstore[['Sales', 'Profit']].agg(['mean', 'max', 'min']))

Analisis Numerik Titanic (Age & Fare)



,Age,Fare
mean,29.699118,32.204208
max,80.000000,512.329200
min,0.420000,0.000000



Analisis Numerik Students Performance (Scores)



,math score,reading score,writing score
mean,66.089,69.169,68.054
max,100.000,100.000,100.000
min,0.000,17.000,10.000



Analisis Numerik Superstore (Sales & Profit)



,Sales,Profit
mean,229.858001,28.656896
max,22638.480000,8399.976000
min,0.444000,-6599.978000


4. Filtering Data

- Lakukan filtering dan analisis data yang menarik dari masing-masing dataset yang digunakan

In [ ]:
print("Titanic Dataset\n")
titanic_filter = df_titanic[(df_titanic['Sex'] == 'female') & (df_titanic['Pclass'] == 1) & (df_titanic['Survived'] == 1)]
print(f"Jumlah penumpang wanita kelas 1 yang selamat: {len(titanic_filter)} orang")
print(garis)
print("StudentPerfomance Dataset\n")
student_filter = df_students[df_students['math score'] == 100]
print(f"Jumlah siswa dengan nilai Matematika sempurna (100): {len(student_filter)} orang")
print("Rata-rata reading score mereka:", student_filter['reading score'].mean())
print(garis)
print("SampleSuperstore Dataset\n")
superstore_filter = df_superstore[(df_superstore['Profit'] < 0) & (df_superstore['Sales'] > 1000)]
print(f"Jumlah transaksi rugi meskipun penjualan di atas $1000: {len(superstore_filter)} transaksi")
print(garis)

Titanic Dataset

Jumlah penumpang wanita kelas 1 yang selamat: 91 orang
StudentPerfomance Dataset

Jumlah siswa dengan nilai Matematika sempurna (100): 7 orang
Rata-rata reading score mereka: 97.85714285714286
SampleSuperstore Dataset

Jumlah transaksi rugi meskipun penjualan di atas $1000: 78 transaksi


## INSIGHT (Wajib dan Paling Penting)

Setiap dataset **WAJIB minimal 3** insight:

Contoh:

- Rata-rata penjualan tertinggi terjadi pada bulan tertentu
- Banyak data kosong pada kolom X
- Sebagian besar data berada pada kategori tertentu

**Titanic Dataset**

Insight 1:
Sebagian besar penumpang tidak selamat, sehingga tingkat survival relatif rendah.

Insight 2:
Penumpang dengan kelas tiket lebih tinggi (Pclass 1) memiliki peluang selamat lebih besar dibanding kelas bawah.

Insight 3:
Terdapat beberapa data kosong pada kolom Age, yang dapat mempengaruhi analisis jika tidak ditangani.


**Students Performance Dataset**

Insight 1:
Sebagian besar siswa memiliki nilai rata-rata yang cukup stabil di antara math, reading, dan writing.

Insight 2:
Siswa yang mengikuti test preparation course cenderung memiliki nilai lebih tinggi.

Insight 3:
Distribusi data cukup seimbang antar kategori seperti gender dan race/ethnicity.


**Superstore Dataset**

Insight 1:
Tidak semua penjualan menghasilkan keuntungan, terlihat dari adanya nilai profit negatif.

Insight 2:
Diskon yang tinggi cenderung menyebabkan penurunan profit.

Insight 3:
Sebagian besar transaksi berasal dari kategori produk tertentu seperti Office Supplies.

## Bandingkan minimal 2 dataset:

1. Dataset mana yang lebih bersih?
2. Dataset mana yang lebih kompleks?
3. Dataset mana yang lebih mudah dianalisis?

**jawaban disini**



1.   Dataset Students Performance lebih bersih karena hampir tidak memiliki data kosong, sedangkan Titanic memiliki beberapa missing value seperti pada kolom Age

2.   Dataset Superstore lebih kompleks karena memiliki banyak variabel seperti produk, lokasi, dan transaksi yang saling berkaitan dibandingkan dataset lainnya.

3.   Dataset Students Performance lebih mudah dianalisis karena strukturnya sederhana dan jumlah variabelnya tidak terlalu banyak.

